Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -254,16 +321,40 @@ open class ChatMessageListVC: _ViewController,
         listView.scrollToMostRecentMessage(animated: animated)
     }
 
-    func updateUnreadMessagesSeparator(at id: MessageId?, previousId: MessageId?) {
+    func updateUnreadMessagesSeparator(at lastReadId: MessageId?) {
+        let previousLastReadId = lastReadMessageId
+        guard previousLastReadId != lastReadId else { return }
+
         func indexPath(for id: MessageId?) -> IndexPath? {
             id.flatMap(getIndexPath)
         }
 
-        let indexPathsToReload = [indexPath(for: previousId), indexPath(for: id)].compactMap { $0 }
+        lastReadMessageId = lastReadId
+        let indexPathsToReload = [indexPath(for: previousLastReadId), indexPath(for: lastReadId)].compactMap { $0 }
+
         guard !indexPathsToReload.isEmpty else { return }
         listView.reloadRows(at: indexPathsToReload, with: .automatic)
     }
 
+    private func isMessageVisible(for messageId: MessageId) -> Bool {
+        guard let indexPath = getIndexPath(forMessageId: messageId) else { return false }
+        return isMessageVisible(for: indexPath)
+    }
+
+    private func isMessageVisible(for indexPath: IndexPath) -> Bool {
```

Comment:
```suggestion
    private func isMessageVisible(at indexPath: IndexPath) -> Bool {
```
Similar as above, just a suggestion to improve the signature readability.

Reply:
Not to self: To be reviewed